materials/material_test.py

```diff
@@ -0,0 +1,79 @@
+"""Unit tests for Material."""
+import unittest
+import numpy as np
+from materials.material import Material, load_from_yaml
+
+class TestMaterial(unittest.TestCase):
+
+    def test_build_properties_simple(self):
+        """Test the build properties function with a simple property."""
+        # Setup
+        yaml_dict = {
+            'density': {
+                'default_value': 1000.,
+                'units': 'kg m^-3',
+                'reference': 'mmpds'
+                }
+            }
+
+        # Action
+        matl = Material('name')
+        matl.build_properties(yaml_dict)
+
+        # Verification
+        self.assertTrue(hasattr(matl, 'density'))
+        self.assertEqual(matl.density.query_value(), 1000.)
+        self.assertEqual(matl.density.units, 'kg m^-3')
+
+    def test_build_properties_1d_table(self):
+        """Test the build properties function with a 1d lookup table."""
+        # Setup
+        dv = 2.0
+        yaml_dict = {
+            'strength': {
+                'default_value': dv,
+                'units': 'MPa',
+                'reference': 'mmpds',
+                'variation_with_state': {
+                    'state_vars': ['temperature'],
+                    'state_vars_units': ['kelvin'],
+                    'value_type': 'multiplier',
+                    'representation': 'table',
+                    'temperature': np.arange(4),
+                    'values': np.arange(4)**2,
+                }
+            }
+        }
+
+        # Action
+        matl = Material('name')
+        matl.build_properties(yaml_dict)
+
+        # Verification
+        self.assertTrue(hasattr(matl, 'strength'))
+        self.assertEqual(matl.strength.units, 'MPa')
+        result = matl.strength.query_value({'temperature': 2})
+        self.assertEqual(result, 4. * dv)
```

Comment:
[note] Cautious here -- the exact equality check will likely fail for any combination of things not a power of 2

I assume you're well aware of that, since you do `almostEqual` elsewhere

Reply:
yes I am being cautious about when I use exact equality checks